atacworks/dl4atac/utils.py

```diff
@@ -525,6 +555,40 @@ def get_intervals(sizesfile, intervalsize, out_dir, val=None,
         df_to_bed(holdout, holdout_file_path)
         return train_file_path, val_file_path, holdout_file_path
 
+    elif regions is not None:
+        # If given regions is a file, then just return the file path
+        if regions.endswith(".bed"):
+            return regions
+        else:
+            final_intervals = pd.DataFrame()
+            regions = regions.strip("[]").split(",")
+            for region in regions:
+                # If regions are specified with intervals like chr1:0-50
+                # Then split the region into chrom and it's range.
+                if region.find(":") != -1:
+                    chrom, chrom_range = region.split(":")
+                    chrom_range = chrom_range.split("-")
+                    chrom_range = [int(value) for value in chrom_range]
+                    intervals = _tile_region_intervals(
```

Comment:
optional: it seems to me that this function can be used in both cases. for the full chromosome case, isn't the range simply "0-chr_length"?

Reply:
Done!